### PR TITLE
Improve dashboard loading

### LINKED ISSUE #4201: [Dashboard] Load days & scheduleentries on in same method as all others
> Nice, maybe it would probably also make sense to load it in `loadAndProcessCollections` because otherwise we wait for everything else.

Network timeline, where some requests are startet simultaneously and the days are loaded after a big group of requests is loaded

_Originally posted by @manuelmeister in https://github.com/ecamp/ecamp3/pull/4135#pullrequestreview-1748800942_

### PR DESCRIPTION
**Pull request by manuelmeister**
This PR introduces optimizations to the dashboard loading process. The modifications in the mounted lifecycle event prioritize loading of essential elements like days, activities, and scheduleEntries. This prioritization enables quicker interactivity, such as making links clickable sooner.

Improvements have also been made to the skeleton loading behavior. These changes ensure that users receive information quickly and meaningfully, improving their overall experience. This approach balances the need for speed with providing complete and useful information to the user.

Closes #4201

### CODE REVIEW

File: frontend/src/components/dashboard/ActivityRow.vue [hunk 1]
```diff
@@ -108,6 +116,15 @@ export default {
   mixins: [dateHelperUTCFormatted],
   props: {
     scheduleEntry: { type: Object, default: () => ({ _meta: { loading: true } }) },
+    loadingEndpoints: {
+      type: [Boolean, Object],
```
Comment (pmattmann) [inline frontend/src/components/dashboard/ActivityRow.vue]: In which case is `loadingEndpoints` assigned a Boolean value?

Comment (manuelmeister) [reply frontend/src/components/dashboard/ActivityRow.vue]: If you don't want to have the loading endpoints behavior at all and don't want to set all endpoints to false.

**Comment by manuelmeister** [comment]
@BacLuc good point to test the flaky-ness of this PR. The tests failed because I did not correctly rename the properties of `loadingEndpoints`

File: frontend/src/components/dashboard/ActivityRow.vue [hunk 2]
```diff
@@ -9,34 +9,28 @@
       </TextAlignBaseline>
       <br />
       <CategoryChip
-        v-if="!scheduleEntry._meta.loading"
+        v-if="
```
Comment (carlobeltrame) [inline frontend/src/components/dashboard/ActivityRow.vue]: Maybe move these new complex v-if conditions to computeds with clear names? E.g. categoryReady or similar? Or invert it and call it categoryLoading or categorySkeleton?

File: frontend/src/components/dashboard/ActivityRow.vue (hunk 1 above)
Comment (carlobeltrame) [reply frontend/src/components/dashboard/ActivityRow.vue]: Couldn't you just set the value to `{}` in that case?

**Review by carlobeltrame — APPROVED** [review]
What you are doing here feels a little like it should be a feature of hal-json-vuex, juggling loading states and Promise.all's etc.
But I don't have a clear mental picture yet of how this would look there. So let's keep it here for now and maybe extract it later.

File: frontend/src/components/dashboard/ActivityRow.vue (hunk 2 above)
Comment (manuelmeister) [reply frontend/src/components/dashboard/ActivityRow.vue]: Simplified them, now that it is only an object

File: frontend/src/components/dashboard/ActivityRow.vue (hunk 1 above)
Comment (manuelmeister) [reply frontend/src/components/dashboard/ActivityRow.vue]: Simplified with just object type

**Review by pmattmann — APPROVED** [review]

**Pull request merged**